Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -2514,3 +2588,66 @@ void vtkSegmentation::CopySegment(vtkSegment* destination, vtkSegment* source, v
     }
   }
 }
+
+//----------------------------------------------------------------------------
+void vtkSegmentation::classInitialize()
+{
+  // Allocate the singleton
+  vtkSegmentationRandomSequence = vtkSegmentation::GetSegmentIDRandomSequenceInstance();
+}
+
+//----------------------------------------------------------------------------
+void vtkSegmentation::classFinalize()
+{
+  vtkSegmentationRandomSequence->Delete();
+  vtkSegmentationRandomSequence = nullptr;
+}
+
+//----------------------------------------------------------------------------
+// Return the single instance of vtkMinimalStandardRandomSequence
+vtkMinimalStandardRandomSequence* vtkSegmentation::GetSegmentIDRandomSequenceInstance()
+{
+  if (!vtkSegmentationRandomSequence)
+  {
+    // Try the factory first
+    vtkSegmentationRandomSequence = (vtkMinimalStandardRandomSequence*)vtkObjectFactory::CreateInstance("vtkMinimalStandardRandomSequence");
+    // if the factory did not provide one, then create it here
+    if (!vtkSegmentationRandomSequence)
+    {
+      vtkSegmentationRandomSequence = vtkMinimalStandardRandomSequence::New();
+    }
+
+    // Get the current ms since the epoch
+    time_t currentTime = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+    int seed = currentTime % 2147483646;
+    vtkSegmentationRandomSequence->SetSeed(currentTime);
+  }
+
+  // return the instance
+  return vtkSegmentationRandomSequence;
+}
+
+//----------------------------------------------------------------------------
+// Implementation of vtkSegmentationRandomSequenceInitialize class.
+//----------------------------------------------------------------------------
+static vtkSegmentationRandomSequenceInitialize* vtkSegmentationRandomSequenceInitializeInstance;
+
+// Must NOT be initialized.  Default initialization to zero is necessary.
+unsigned int vtkSegmentationRandomSequenceInitialize::Count;
+
+vtkSegmentationRandomSequenceInitialize::vtkSegmentationRandomSequenceInitialize()
```

Comment:
We could move `vtkSingleton.h`[^1] into `vtkAddon`[^2] or `MRML/Core` and use it here.

[^1]: https://github.com/Slicer/Slicer/blob/main/Libs/MRML/DisplayableManager/vtkSingleton.h
[^2]: https://github.com/Slicer/vtkAddon

Reply:
Agreed

Reply:
Thanks, PR created [here](https://github.com/Slicer/vtkAddon/pull/49) and I've updated this PR to use vtkSingleton macros.